templates/master/00-master/two-node-with-fencing/files/fencing-validator.yaml

```diff
@@ -0,0 +1,520 @@
+            TRANSPORT=ssh; CONDUCTOR="$IP_B"; _sudo_check; log "Using SSH; conductor=$CONDUCTOR"
+          elif ocdebug_ok "$NODE_A" && ocdebug_ok "$NODE_B"; then
+            TRANSPORT=ocdebug; CONDUCTOR="$NODE_B"; log "Using oc debug; conductor=$CONDUCTOR"
+          else
+            log_err "Auto mode: neither SSH nor oc debug reachable on both nodes."; exit $EXIT_GENERIC
+          fi
+          ;;
+        ssh)
+          (ssh_ok "$IP_A" && ssh_ok "$IP_B") || { log_err "SSH not available to both nodes"; exit $EXIT_GENERIC; }
+          CONDUCTOR="$IP_B"; _sudo_check; log "Using SSH; conductor=$CONDUCTOR"
+          ;;
+        ocdebug)
+          (ocdebug_ok "$NODE_A" && ocdebug_ok "$NODE_B") || { log_err "oc debug not available on both nodes"; exit $EXIT_GENERIC; }
+          CONDUCTOR="$NODE_B"; log "Using oc debug; conductor=$CONDUCTOR"
+          ;;
+      esac
+
+      # -------- Pacemaker names --------
+      PCMK_A="$NODE_A"
+      PCMK_B="$NODE_B"
+      log "Mapping: $NODE_A -> $(short_hostname "$PCMK_A") ; $NODE_B -> $(short_hostname "$PCMK_B")"
+    
+      # -------- Run --------
+      log "Mode: transport=$TRANSPORT disruptive=$DISRUPTIVE dry-run=$DRY_RUN"
+      log "=== Non-disruptive validation ==="
+      check_stonith || exit $EXIT_STONITH_MISSING
+      if ! ( pcmkonline "$PCMK_A" && pcmkonline "$PCMK_B" ); then
+        log_err "Both nodes must be ONLINE (Pacemaker)"
+        exit $EXIT_PACEMAKER_OFFLINE
+      fi
+      log_ok "Both nodes ONLINE"
+      check_daemon_status || exit $EXIT_DAEMONS_BAD
+      etcd_wait
+      check_fencing_secret_bindings || exit $EXIT_FENCING_SECRETS_MISMATCH
+      log_ok "[PASS] Non-disruptive checks complete"
+    
+      if ! $DISRUPTIVE; then
+        $DRY_RUN && dry_run_plan
+        echo "Done (non-disruptive)."
+        exit $EXIT_OK
+      fi
+    
+      $DRY_RUN && { dry_run_plan; exit $EXIT_OK; }
+    
+      log "=== Disruptive validation ==="
+      # Fence A
+      switch_conductor_for "$NODE_A"
+      log "Fencing $NODE_A (PCMK: $PCMK_A)"
+      fence "$PCMK_A" || exit $EXIT_GENERIC
+      wait_not_ready "$NODE_A"; wait_ready "$NODE_A"; etcd_wait; check_daemon_status || exit $EXIT_DAEMONS_BAD
+    
+      # Fence B
+      switch_conductor_for "$NODE_B"
+      log "Fencing $NODE_B (PCMK: $PCMK_B)"
+      etcd_wait
+      fence "$PCMK_B" || exit $EXIT_GENERIC
+      wait_not_ready "$NODE_B"; wait_ready "$NODE_B"; etcd_wait; check_daemon_status || exit $EXIT_DAEMONS_BAD
+    
+      log_ok "Disruptive validation PASSED"
+    }
+    
+    # ================================
+    # helper functions
+    # ================================
+    
+    log() {
+      printf '\033[36m[INFO]\033[0m %s\n' "$*"
+    }
+    log_warn() {
+      printf '\033[33m[WARN]\033[0m %s\n' "$*"
+    }
+    log_err() {
+      printf '\033[31m[ERROR]\033[0m %s\n' "$*" >&2
+    }
+    log_ok() {
+      printf '\033[32m[OK]\033[0m %s\n' "$*"
+    }
+    
+    _is_ipv6() {
```

Comment:
I'm not sure I follow this pattern of some functions starting with underscores, let's keep them consistent and remove the beginning underscore. Unless there's a purpose here, if so we should make sure it's clear